src/Components/ServiceScene/Breakdowns/PatternsViewTextSearch.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  CustomVariable,
+  SceneComponentProps,
+  sceneGraph,
+  SceneObjectBase,
+  SceneObjectState,
+  SceneVariableSet,
+} from '@grafana/scenes';
+import React, { ChangeEvent } from 'react';
+import { Field, Icon, Input } from '@grafana/ui';
+import { css } from '@emotion/css';
+import { PATTERNS_TEXT_FILTER } from '../../../services/variables';
+import { PatternsBreakdownScene } from './PatternsBreakdownScene';
+import { debouncedFuzzySearch } from '../../../services/search';
+
+export interface PatternsViewTextSearchState extends SceneObjectState {
+  patternFilter?: string;
+}
+
+export class PatternsViewTextSearch extends SceneObjectBase<PatternsViewTextSearchState> {
+  public static Component = PatternTextSearchComponent;
+
+  constructor(state: PatternsViewTextSearchState) {
+    super({
+      ...state,
+      patternFilter: state.patternFilter,
+      $variables: new SceneVariableSet({
+        variables: [new CustomVariable({ name: PATTERNS_TEXT_FILTER, value: state.patternFilter ?? '' })],
+      }),
+    });
+
+    this.addActivationHandler(this.onActivate.bind(this));
+  }
+
+  /**
+   * On click callback to clear current text search
+   */
+  public clearSearch = () => {
+    this.setState({
+      patternFilter: undefined,
+    });
+  };
+
+  /**
+   * Search input onchange callback
+   * @param e
+   */
+  public handleSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      patternFilter: e.target.value,
+    });
+  };
+
+  /**
+   * Activation handler
+   * @private
+   */
+  private onActivate() {
+    this._subs.add(
+      this.subscribeToState((newState, prevState) => {
+        if (newState.patternFilter !== prevState.patternFilter) {
+          const parent = sceneGraph.getAncestor(this, PatternsBreakdownScene);
+          if (parent.state.patternFrames) {
+            debouncedFuzzySearch(
+              parent.state.patternFrames.map((frame) => frame.pattern),
+              this.state.patternFilter ?? '',
+              this.onSearchResult
```

Comment:
I find this structured in an unexpected way. Why passing a callback instead of 

```suggestion
            this.onSearchResult(
              debouncedFuzzySearch(
                parent.state.patternFrames.map((frame) => frame.pattern),
                this.state.patternFilter ?? '',
              )
            );
```

The reason is not stilistic, but mainly for readability. I would directly know that `this.onSearchResult` is called with the output of `debouncedFuzzySearch`, instead of inferring or having to check that `debouncedFuzzySearch ` internally calls `this.onSearchResult()`.

Reply:
Right, makes sense. Thanks for the clarification.